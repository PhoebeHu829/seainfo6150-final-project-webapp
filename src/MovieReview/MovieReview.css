.form {
  display: flex;
  flex-direction: column;
  margin-top: 30px;
}

.review_content {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  font-family: 'Patrick Hand', cursive;
  margin-left: -25%;
  margin-top: -10%;
  height: 100vh;
  background-image: url('https://data.photofunky.net/output/image/e/1/b/a/e1ba35/photofunky.gif');
  background-repeat: no-repeat;
  background-position: right bottom;
  font-size: 20px;
}

.form label {
  margin: 5px;
}

.overview {
  padding: 20px;
}

.overview > label {
  padding-bottom: 20px;
}

.overview div,
.submit_review svg {
  margin-top: 5px;
  margin-bottom: 5px;
  color: rgb(189, 145, 23);
}

.overview div:hover {
  color: rgb(58, 5, 156);
}

#myTextId {
  width: 100%;
  margin: 10px;
  border-radius: 5px;
  height: 100px;
  border-color: rgb(58, 5, 156);
}

#myDropdown {
  margin: 10px;
  border-radius: 5px;
  border-color: rgb(58, 5, 156);
  height: 40px;
  padding: 5px;
  font-family: 'Patrick Hand', cursive;
  font-size: 18px;
}

.button,
.submit_review button {
  width: 50%;
  border-radius: 15px;
  padding: 10px;
  margin: 10px;
  font-family: 'Patrick Hand', cursive;
  border-width: 0px;
  background-color: rgb(58, 5, 156);
  color: azure;
}

.review_content img {
  width: 180px;
  height: 240px;
  margin: 100px;
  border-radius: 5px;
  box-shadow: 1px 1px 50px #888888;
}

.review_content img:hover {
  box-shadow: 1px 1px 50px #f16969;
}

@media only screen and (max-width: 1000px) {
  .review_content {
    background-image: none;
  }

  .review_content img {
    margin-right: 10px;
  }
}

@media only screen and (max-width: 750px) {
  .review_content img {
    display: none;
  }

  .form,
  #myDropdown {
    display: flex;
    width: 60%;
    font-size: 15px;
  }
}
