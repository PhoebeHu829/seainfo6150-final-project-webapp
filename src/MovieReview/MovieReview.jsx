import React, { useState, useEffect } from 'react';
import './MovieReview.css';
import { Link, useHistory } from 'react-router-dom';
import { FaStar } from 'react-icons/fa';

const MoviewReview = ({ data, id, setSubmittedForm, submittedForm }) => {
  const history = useHistory({});
  const [movie_review, change_movie] = useState(id);
  const movies = data.results;
  const option_list = movies.map((m) => (
    <option value={m.id} key={m.id}>
      {' '}
      {m.title}
    </option>
  ));
  const path_map = new Map();
  const movie_title_map = new Map();
  movies.forEach((m) => {
    path_map.set(m.id, m.poster_path);
    movie_title_map.set(m.id, m.title);
  });

  const handleReviewAgain = (event) => {
    setSubmittedForm();
  };

  useEffect(() => {
    setSubmittedForm();
  }, [id, setSubmittedForm]);

  function onSubmit(e) {
    e.preventDefault();
    const data = new FormData(e.target);
    const iterator = data.values();
    const general_review = iterator.next().value;
    let general_review_back;
    switch (general_review) {
      case '4': {
        general_review_back = (
          <span>
            <FaStar />
            <FaStar />
            <FaStar />
            <FaStar />
          </span>
        );
        break;
      }
      case '3': {
        general_review_back = (
          <span>
            <FaStar />
            <FaStar />
            <FaStar />
          </span>
        );
        break;
      }
      case '2': {
        general_review_back = (
          <span>
            <FaStar />
            <FaStar />
          </span>
        );
        break;
      }
      default:
        general_review_back = (
          <span>
            <FaStar />
          </span>
        );
    }

    const detail_review = iterator.next().value.trim();
    setSubmittedForm({
      general: general_review_back,
      detail: detail_review,
      title: movie_title_map.get(parseInt(movie_review)),
    });
  }

  return (
    <div className="review_content">
      <Link to={'/movie/' + movie_review}>
        <img
          src={
            'https://www.themoviedb.org/t/p/w600_and_h900_bestv2/' +
            path_map.get(parseInt(id), 10)
          }
          alt="poster"
        />
      </Link>
      <form onSubmit={onSubmit} className="form">
        <label htmlFor="myDropdownId">Which moview you want to review?</label>
        <div>
          <select
            id="myDropdown"
            value={id}
            onChange={(event) => {
              change_movie(event.target.value);
              history.push('/movieReview/' + event.target.value);
            }}
          >
            {option_list}
          </select>
        </div>
        {submittedForm ? (
          <div className="submit_review">
            {' '}
            <h2>Thank you review {submittedForm.title}</h2>
            {submittedForm.general}
            <p> {submittedForm.detail}</p>
            <button onClick={handleReviewAgain}> Review Again</button>
          </div>
        ) : (
          <>
            <div className="overview">
              <div>
                <input
                  type="radio"
                  name="overview_rate"
                  id="nice"
                  value="4"
                  required
                />
                <label htmlFor="nice">
                  <FaStar />
                  <FaStar />
                  <FaStar />
                  <FaStar />
                </label>
              </div>
              <div>
                <input type="radio" name="overview_rate" id="good" value="3" />
                <label htmlFor="good">
                  <FaStar />
                  <FaStar />
                  <FaStar />
                </label>
              </div>
              <div>
                <input type="radio" name="overview_rate" id="ok" value="2" />
                <label htmlFor="ok">
                  <FaStar />
                  <FaStar />
                </label>
              </div>
              <div>
                <input type="radio" name="overview_rate" id="bad" value="1" />
                <label htmlFor="bad">
                  <FaStar />
                </label>
              </div>
            </div>
            <label htmlFor="myTextId">Write down your review:</label>
            <textarea type="text" name="myText" id="myTextId" required />
            <input type="submit" value="Submit" className="button" />
          </>
        )}
      </form>
      )}
    </div>
  );
};

export default MoviewReview;
