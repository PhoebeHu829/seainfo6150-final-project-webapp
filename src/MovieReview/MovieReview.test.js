import React from 'react';
import MovieReview from './MovieReview.jsx';

const data = {
  page: 1,
  results: [
    {
      adult: false,
      backdrop_path: '/9yBVqNruk6Ykrwc32qrK2TIE5xw.jpg',
      genre_ids: [14, 28, 12, 878, 53],
      id: 460465,
      original_language: 'en',
      original_title: 'Mortal Kombat',
      overview: '.',
      popularity: 4901.131,
      poster_path: '/2xmx8oPlbDaxTjHsIOZvOt5L3aJ.jpg',
      release_date: '2021-04-07',
      title: 'Mortal Kombat',
      video: false,
      vote_average: 8.1,
      vote_count: 980,
    },
  ],
};

it('renders the MovieReview component correctly', () => {
  const { container } = render(
    <MovieReview
      data={data}
      id={'399655'}
      submittedForm={[]}
      setSubmittedForm={() => {}}
    />,
  );
  expect(container).toMatchSnapshot();
});
