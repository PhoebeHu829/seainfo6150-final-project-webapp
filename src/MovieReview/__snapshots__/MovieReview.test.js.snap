// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the MovieReview component correctly 1`] = `
<div>
  <div
    class="review_content"
  >
    <a
      href="/movie/399655"
    >
      <img
        alt="poster"
        src="https://www.themoviedb.org/t/p/w600_and_h900_bestv2/undefined"
      />
    </a>
    <form
      class="form"
    >
      <label
        for="myDropdownId"
      >
        Which moview you want to review?
      </label>
      <div>
        <select
          id="myDropdown"
        >
          <option
            value="460465"
          >
             Mortal Kombat
          </option>
        </select>
      </div>
      <div
        class="submit_review"
      >
         
        <h2>
          Thank you review 
        </h2>
        <p>
           
        </p>
        <button>
           Review Again
        </button>
      </div>
    </form>
    )}
  </div>
</div>
`;
